contracts/contracts/strategies/NativeStaking/CompoundingStakingSSVStrategy.sol

```diff
@@ -0,0 +1,286 @@
+    function initialize(
+        address[] memory _rewardTokenAddresses,
+        address[] memory _assets,
+        address[] memory _pTokens
+    ) external onlyGovernor initializer {
+        InitializableAbstractStrategy._initialize(
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+
+        // Approves the SSV Network contract to transfer SSV tokens for deposits
+        IERC20(SSV_TOKEN).approve(SSV_NETWORK, type(uint256).max);
+    }
+
+    /// @notice Unlike other strategies, this does not deposit assets into the underlying platform.
+    /// It just checks the asset is WETH and emits the Deposit event.
+    /// To deposit WETH into validators `registerSsvValidator` and `stakeEth` must be used.
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    /// @param _asset Address of asset to deposit. Has to be WETH.
+    /// @param _amount Amount of assets that were transferred to the strategy by the vault.
+    function deposit(address _asset, uint256 _amount)
+        external
+        override
+        onlyVault
+        nonReentrant
+    {
+        require(_asset == WETH, "Unsupported asset");
+        depositedWethAccountedFor += _amount;
+        _deposit(_asset, _amount);
+    }
+
+    /// @dev Deposit WETH to this strategy so it can later be staked into a validator.
+    /// @param _asset Address of WETH
+    /// @param _amount Amount of WETH to deposit
+    function _deposit(address _asset, uint256 _amount) internal {
+        require(_amount > 0, "Must deposit something");
+        /*
+         * We could do a check here that would revert when "_amount % 32 ether != 0". With the idea of
+         * not allowing deposits that will result in WETH sitting on the strategy after all the possible batches
+         * of 32ETH have been staked.
+         * But someone could mess with our strategy by sending some WETH to it. And we might want to deposit just
+         * enough WETH to add it up to 32 so it can be staked. For that reason the check is left out.
+         *
+         * WETH sitting on the strategy won't interfere with the accounting since accounting only operates on ETH.
+         */
+        emit Deposit(_asset, address(0), _amount);
+    }
+
+    /// @notice Unlike other strategies, this does not deposit assets into the underlying platform.
+    /// It just emits the Deposit event.
+    /// To deposit WETH into validators `registerSsvValidator` and `stakeEth` must be used.
+    /// Will NOT revert if the strategy is paused from an accounting failure.
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        uint256 newWeth = wethBalance - depositedWethAccountedFor;
+
+        if (newWeth > 0) {
+            depositedWethAccountedFor = wethBalance;
+
+            _deposit(WETH, newWeth);
+        }
+    }
```

Comment:
This function seems a bit asymmetrical in comparison with the simple deposit. 
In this `depositAll()` we kind of ensure that the strategy have the deposited amount, while on the `deposit()` we don't check at all that we have the correct amount of token. 

Maybe we should ensure that we have at least amount deposited in the `deposit()` function?